File: src-1.x/BYTES.NET/WPF/Converters/CountToVisibilityConverter.vb
'import .net namespace(s) needed
Imports System.Globalization

Namespace WPF.Converters

    ''' <summary>
    ''' converter returning the visibility state by the items count of a given object
    ''' </summary>
    Public Class CountToVisibilityConverter

        Implements IValueConverter

        Public Function Convert(value As Object, targetType As Type, parameter As Object, culture As CultureInfo) As Object Implements IValueConverter.Convert

            If IsNothing(parameter) Then

                Try

                    If value.Count >= 1 Then 'check for 'Count'

                        Return Visibility.Visible

                    Else

                        Return Visibility.Collapsed

                    End If

                Catch ex As Exception

                    Try

                        If value.length >= 1 Then 'check for 'Length' (e.g. in arrays)

                            Return Visibility.Visible

                        Else

                            Return Visibility.Collapsed

                        End If

                    Catch exer As Exception

                        Return Visibility.Collapsed

                    End Try

                End Try

            Else

                Try

                    If value.count >= CInt(parameter) Then 'check for 'Count'

                        Return Visibility.Visible

                    Else

                        Return Visibility.Collapsed

                    End If

                Catch ex As Exception

                    Try

                        If value.length >= CInt(parameter) Then 'check for 'Length' (e.g. in arrays)

                            Return Visibility.Visible

                        Else

                            Return Visibility.Collapsed

                        End If

                    Catch exer As Exception

                        Return Visibility.Collapsed

                    End Try

                End Try

            End If

        End Function

        Public Function ConvertBack(value As Object, targetType As Type, parameter As Object, culture As CultureInfo) As Object Implements IValueConverter.ConvertBack

            Return Binding.DoNothing

        End Function

    End Class

End Namespace

File: src-2023/BYTES.NET.WPF/Converters/CountToVisibilityConverter.cs
//import .net namespace(s) required
using System;
using System.Collections.Generic;
using System.Globalization;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows;
using System.Windows.Data;

namespace BYTES.NET.WPF.Converters
{
    public class CountToVisibilityConverter : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
        {
            try
            {
                if (parameter == null)
                {
                    // No parameter provided, check for Count or Length >= 1
                    if (value is System.Collections.ICollection collection && collection.Count >= 1)
                    {
                        return Visibility.Visible;
                    }
                    else if (value != null && value.GetType().IsArray && ((Array)value).Length >= 1)
                    {
                        return Visibility.Visible;
                    }
                    else
                    {
                        return Visibility.Collapsed;
                    }
                }
                else
                {
                    // Parameter provided, check for Count or Length >= parameter
                    int paramValue;
                    if (int.TryParse(parameter.ToString(), out paramValue))
                    {
                        if (value is System.Collections.ICollection collection && collection.Count >= paramValue)
                        {
                            return Visibility.Visible;
                        }
                        else if (value != null && value.GetType().IsArray && ((Array)value).Length >= paramValue)
                        {
                            return Visibility.Visible;
                        }
                        else
                        {
                            return Visibility.Collapsed;
                        }
                    }
                    else
                    {
                        return Visibility.Collapsed;
                    }
                }
            }
            catch (Exception ex)
            {
                return Visibility.Collapsed;
            }
        }

        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
        {
            return Binding.DoNothing;
        }
    }
}
